### PR TITLE
Watch library directories with perm errors

### LINKED ISSUE #7130: [Issue]: Directory watch disabled if subdirectory (e.g. lost+found) is unreadable
### Please describe your bug

This morning I've reviewed my jellyfin logs for the first time.  I've been able to correct some permission issues and get them quite quiet. But this remains:

```
jan. 07 09:40:15 gont jellyfin[3320044]: System.UnauthorizedAccessException: Access to the path '/space/dvds/lost+found' is denied.
jan. 07 09:40:15 gont jellyfin[3320044]:  ---> System.IO.IOException: Permission denied
jan. 07 09:40:15 gont jellyfin[3320044]:    --- End of inner exception stack trace ---
jan. 07 09:40:15 gont jellyfin[3320044]: [09:40:15] [INF] Stopping directory watching for path /space/dvds
```

The lost+found directory should not be readable to non-admins and should in any case not be catalogued. It seems unnecessary to let a permission denied exception disable watching a directory hierarchy.

Regards,
  Nicolai

### Jellyfin Version

10.7.7

### if other:

_No response_

### Environment

```markdown
- OS: Linux, Ubuntu 20.04.3 LTS
- Virtualization: Running directly on host, not in VM or container
- Clients: This is a storage backend issue
- Browser: This is a storage backend issue
- FFmpeg Version: This is a storage backend issue
- Playback Method: This is a storage backend issue
- Hardware Acceleration: This is a storage backend issue
- Plugins: This is a storage backend issue
- Reverse Proxy: This is a storage backend issue
- Base URL: This is a storage backend issue
- Networking: This is a storage backend issue
- Storage: Local disk, ext4 filesystem
```

### Jellyfin logs

```shell
jan. 07 09:40:15 gont jellyfin[3320044]: [09:40:15] [ERR] Error in Directory watcher for: /space/dvds
jan. 07 09:40:15 gont jellyfin[3320044]: System.UnauthorizedAccessException: Access to the path '/space/dvds/lost+found' is denied.
jan. 07 09:40:15 gont jellyfin[3320044]:  ---> System.IO.IOException: Permission denied
jan. 07 09:40:15 gont jellyfin[3320044]:    --- End of inner exception stack trace ---
jan. 07 09:40:15 gont jellyfin[3320044]: [09:40:15] [INF] Stopping directory watching for path /space/dvds
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bhowe34**
**Changes**
Continue to watch library directories if a subdirectory has invalid permissions.

An error message is printed but the watcher is not stopped so other subdirectories can still be monitored.
Error message:
```
[23:03:24] [ERR] [8] Emby.Server.Implementations.IO.LibraryMonitor: Permission error for Directory watcher: /tmp/jf-test2
System.UnauthorizedAccessException: Access to the path '/tmp/jf-test2' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
```

**Issues**
Fixes #7130

### CODE REVIEW

File: Emby.Server.Implementations/IO/LibraryMonitor.cs [hunk 1]
```diff
@@ -330,6 +330,12 @@ private void OnWatcherError(object sender, ErrorEventArgs e)
             var ex = e.GetException();
             var dw = (FileSystemWatcher)sender;
 
+            if (ex.GetType() == typeof(UnauthorizedAccessException))
+            {
+                _logger.LogError(ex, "Permission error for Directory watcher: {Path}", dw.Path);
+                return;
+            }
```
Comment (crobibero) [inline Emby.Server.Implementations/IO/LibraryMonitor.cs:337]:
```suggestion
            if (ex is UnauthorizedAccessException unauthorizedAccessException)
            {
                _logger.LogError(unauthorizedAccessException, "Permission error for Directory watcher: {Path}", dw.Path);
                return;
            }
```

**Review by 1337joe — APPROVED** [review]

**Comment by JPVenson** [comment]
Does this lead to an error printed each time something changes in that subdirectory?

**Comment by bhowe34** [comment]
From the doc

> Error Occurs when the instance of [FileSystemWatcher](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher?view=net-8.0) is unable to continue monitoring changes or when the internal buffer overflows.

In my testing it only prints the error when the watcher starts and then the watcher ignores that subdirectory. If the permissions for the directory do change to allow access the watcher would need to be restarted which happens when a manual library scan is triggered.

**Review by Gauvino — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

**Comment by evert** [comment]
Pumped about this! thank you

**Pull request merged**